# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/gtk/DMABufLibWrapper.cpp
# Commit: fe7075f48cf0
# Full Hash: fe7075f48cf031d5686b3268a0e738581721b386
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2020-07-23 03:20:25
# Description:
#   Bug 1650748 - Prevent DMABUF usage with GLX. r=nical
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D84256
# ==============================================================================

diff -r 36b357364a21 -r fe7075f48cf0 widget/gtk/DMABufLibWrapper.cpp
--- a/widget/gtk/DMABufLibWrapper.cpp	Wed Jul 22 21:00:42 2020 +0000
+++ b/widget/gtk/DMABufLibWrapper.cpp	Wed Jul 22 18:58:11 2020 +0000
@@ -8,6 +8,7 @@
 #include "DMABufLibWrapper.h"
 #include "mozilla/StaticPrefs_widget.h"
 #include "mozilla/StaticPrefs_media.h"
+#include "mozilla/gfx/gfxVars.h"
 
 #include <sys/types.h>
 #include <sys/stat.h>
@@ -161,14 +162,16 @@
 
   mIsDMABufConfigured = true;
   bool isDMABufUsed =
+      gfx::gfxVars::UseEGL() &&
+      (
 #ifdef NIGHTLY_BUILD
-      StaticPrefs::widget_wayland_dmabuf_basic_compositor_enabled() ||
-      StaticPrefs::widget_dmabuf_textures_enabled() ||
+          StaticPrefs::widget_wayland_dmabuf_basic_compositor_enabled() ||
+          StaticPrefs::widget_dmabuf_textures_enabled() ||
 #endif
-      StaticPrefs::widget_dmabuf_webgl_enabled() ||
-      StaticPrefs::media_ffmpeg_dmabuf_textures_enabled() ||
-      StaticPrefs::media_ffmpeg_vaapi_enabled() ||
-      StaticPrefs::media_ffmpeg_vaapi_drm_display_enabled();
+          StaticPrefs::widget_dmabuf_webgl_enabled() ||
+          StaticPrefs::media_ffmpeg_dmabuf_textures_enabled() ||
+          StaticPrefs::media_ffmpeg_vaapi_enabled() ||
+          StaticPrefs::media_ffmpeg_vaapi_drm_display_enabled());
 
   if (!isDMABufUsed) {
     // Disabled by user, just quit.
